GeneratorInterface/Hydjet2Interface/test/Hydjet2Analyzer.cc

```diff
@@ -524,22 +582,27 @@ void Hydjet2Analyzer::analyze(const edm::Event &iEvent, const edm::EventSetup &i
           float e = (*it)->momentum().e();
           float pseudoRapidity = (*it)->momentum().pseudoRapidity();
           int charge = -1;
+          float mass = -1.;
           if ((pdt->particle(pdg_id))) {
             part = pdt->particle(pdg_id);
             charge = static_cast<int>(part->charge());
+            mass = (part->mass());
           }
-
-          /*
-          if(pdg_id==-130){ //there are not -130 in pdt
-	    part = pdt.particle(130);
-	  }else if(!(pdt.particle(pdg_id))){ //skip if not in PDT!!!
-	    cout<<" Out of PDT: "<< pdg_id<<endl;
-	    continue;
-          }else{
-	    part = pdt.particle(pdg_id);
+          if (mass < 0) {
+            if ((abs(pdg_id) == 130) || (abs(pdg_id) == 310)) {
+              mass = 0.497611;
+            } else {
+              //cout<<"Error! Mass for "<< pdg_id <<" not found in PDT!!!"<<endl;
```

Comment:
Does it makes sense to continue if there is a negative mass here?

Reply:
We using mass only for ET, so with negative mass we still can use other histograms. Anyway this analyser is used for tests and tuning only.